### PR TITLE
New feature: solarsystem

### PR DESCRIPTION
**Pull request by mommermi**
I present a new feature: `solarsystem`

Currently, only queries to the JPL Horizons service are implemented, enabling the user to obtain ephemerides, orbital elements, and state vectors for basically all known Solar System objects. The idea is to implement several other services under `solarsystem` that are relevant to Solar System researchers (access to Minor Planet Center databases, JPL SBDB, IMCCE, AstDyS, NEODyS...)

The currently available functions in `JPLClass` produce `astropy.table` objects as output. I would love to use `Qtable`s instead, but I am having problems doing so. For now, simple `Table` objects work just fine and I will implement `Qtable`s later.

Full documentation and working tests are provided with this PR.

### CODE REVIEW

**Comment by keflavich** [comment]
General 👍 , thanks for submitting!  I haven't had time to review yet.  @mommermi, do you have any ideas for someone who might be particularly qualified to review the PR?

**Comment by mommermi** [comment]
Thanks! @migueldvb is familiar with both JPL Horizons and the precursor ([callhorizons](https://github.com/mommermi/callhorizons/)) of this submodule. He is definitely qualified, but I don't know if he is available.

**Comment by migueldvb** [comment]
Sure! It is great to have this functionality in astroquery. I will take a look at the PR.

**Comment by migueldvb** [comment]
Thank you for including the suggested changes in mommermi/astroquery/pull/1 - mommermi/astroquery/pull/5. It seems that all the queries work as the excellent documentation says they should and I only have a few minor comments on the module listed below.

One general comment is that this service could be renamed to `horizons` and other services for Solar System researchers that are mentioned in the pull request (MPC, SBDB, IMCCE...) could be implemented as different modules in astroquery for simplicity, since they have different APIs and there would be no shared code between them. Perhaps @keflavich and @bsipocz will have some suggestion about this?

Some other things:

- The docstring of JPLClass says that `id_type` default value is `any`, but it is actually `smallbody`. It seems that `any` is not supported at the moment so it could be that just the docstring needs to be updated?

- The docstring for ephemerides, elements and vectors methods says that "This method requires the ``id``, ``location``, and ``epochs`` parameters of the ``JPL`` object to be set.", but the `location` will default to geocentric or heliocentric, respectively and `epochs` to current date if not set. This is already explained in the documentation and JPLClass so I think it is just a matter of modifying the first sentence of the ephemerides and vectors docstring.

- The changelog entry in the CHANGES file for this new service may need to be rebased on current master since there has been some new entries in the changelog since this pull request was open.

It will be great to have this functionality in the next release!

**Comment by mommermi** [comment]
Thanks, @migueldvb , for your detailed and helpful review! I took care of your final suggestions with the latest commit. Please note that I moved the announcement of this tool to v0.8.3.dev in the CHANGES file.

Concerning the naming and organization, I am open for suggestions. My motivation for 'solarsystem' was to have all solar system related things grouped together, so that people could call different services like
```
from astroquery.solarsystem import JPL
from astroquery.solarsystem import MPC
# etc.
```
But I am also fine creating a separate module for each service - @keflavich and @bsipocz, you decide!

File: CHANGES [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+
+- New tool: JPL Horizons service to obtain ephemerides, orbital elements, and
+  state vectors for Solar System objects.
+
+
+0.3.7 (2018-01-25)
 ------------------
 
 - New tool: Exoplanet Orbit Catalog, NASA Exoplanet Archive [#771]
-- MAST: Added convenience function to list available missions. [#947]
```
Comment (bsipocz) [inline CHANGES]: Could you please do a rebase to remove these unrelated changes?  I'll be able to do a full review later next week.

Comment (mommermi) [reply CHANGES]: Thanks! Please let me know if the latest commit does what you want. If not please be more specific - I am not yet really familiar with the term 'rebase`. Do you want me to remove all unrelated changes? Sorry.

**Comment by mommermi** [comment]
Thanks, @keflavich! I renamed the submodule to ``JPLHorizons`` as you and @migueldvb  suggested and did the rebase (hopefully correctly). If there is anything left for me to do, please let me know!

**Comment by keflavich** [comment]
Sorry about the delayed reply.  It looks like there are some conflicts that need to be resolved now in the changelog, and I think one of @bsipocz's recent changes might have fixed our CI issues.  Could you fix the changelog conflict and update the PR?  Then, hopefully, we'll see travis go green and be able to merge.

File: astroquery/jplhorizons/core.py [hunk 2]
```diff
@@ -0,0 +1,971 @@
...
+
+
+        Examples
+        --------
+            >>> from astroquery.jplhorizons import Horizons
+            >>> eros = Horizons(id='433', location='568',
+            ...              epochs={'start':'2017-01-01',
+            ...                      'stop':'2017-02-01',
+            ...                      'step':'1d'})
+            >>> print(eros) # doctest: +SKIP
```
Comment (keflavich) [inline astroquery/jplhorizons/core.py]: The trailing whitespace on this line is causing tests to fail

**Comment by mommermi** [comment]
Thanks, @keflavich! I modified the tests as you suggested and fixed some other minor things in #1051. Thanks for your help with the PEP8 issues. Is there a list of the issues relevant to `astropy`? Some like E501 and E129 don't seem to matter to Travis CI, but others do. It would be useful to have a list of those issues that really matter.

**Comment by keflavich** [comment]
Good question.  @bsipocz, do you know if such a list exists?

I don't know, but looking at the travis pep8 report should always tell you.

**Comment by bsipocz** [comment]
We have a list in `setup.cfg`, those are the only ones triggering a fail on travis. If you run `pycodestyle` locally, it would also only complain about those, but we only have a very limited list for the pep8speaks bot, so it will definitely complain more.

Actually in astropy core we removed the list from the setup file, and hard wired it into the travis config, we can pick up more stuff locally yet not enforcing more strict rules on all contributors.

**Pull request merged**